### DISCUSSION TITLE
[Pytest Failure]- Import error failed to import .pyb, .pyc,.pyx file

### DISCUSSION THREAD

**Discussion by swagaths1** [explosion/spaCy#12253] [Help: Installation] [answered]
Not able to run any pytest facing the following error
E   ModuleNotFoundError: No module named 'spacy.pipeline.pipe'
Basically its not able to find and .pyx, .pyb, .pxd files.

pytest -v test_architectures.py
ImportError while loading conftest '/Users/swagath/Documents/GitHub/spaCy/spacy/tests/conftest.py'.
../__init__.py:14: in <module>
    from . import pipeline  # noqa: F401
../pipeline/__init__.py:1: in <module>
    from .attributeruler import AttributeRuler
../pipeline/attributeruler.py:6: in <module>
    from .pipe import Pipe
E   ModuleNotFoundError: No module named 'spacy.pipeline.pipe'

 
Steps to Reproduce:

step1: go to test case folder of spacy
            cd spaCy/spacy/tests
step2: run any pytest 
         eg:
          pytest -v test_architectures.py

* Operating System: linux
* Flatform : x86
* Python Version Used: any
* spaCy Version Used: 2.3.5 and lower versions as well

**Comment by rmitsch** [comment]
Hi @swagaths1, please specify how exactly you installed spaCy.

**Comment by swagaths1** [comment]
This is not issue with installation of spacy

Please find the steps below how I installed spacy


step 1: I installed srsly by using below command
                pip install srsly  

step 2: I installed thinc by using below command
                pip install thinc

step 3: I installed spacy by using below command
                pip install spacy==2.3.5


After this steps Im able to see all the dependencies in my pip freeze list
<img width="1728" alt="Screenshot 2023-02-09 at 11 12 47 AM" src="https://user-images.githubusercontent.com/121920450/217728139-15640a84-bba2-425b-93d8-616c98a13980.png">

Even Im able to import spacy
<img width="1728" alt="Screenshot 2023-02-09 at 11 08 15 AM" src="https://user-images.githubusercontent.com/121920450/217728183-d7073f7b-2733-4507-97f8-adf2f9ed0074.png">

The issue is only while running pytest

**Comment by adrianeboyd** [comment]
Run the tests like this instead:

```bash
cd /some/other/directory
pytest --pyargs spacy.tests.test_architectures
```

Running the tests from within the `tests/` directory only works with an editable install.

**Comment by swagaths1** [comment, marked as answer]
I'm getting following error 
<img width="1691" alt="Screenshot 2023-02-09 at 1 42 29 PM" src="https://user-images.githubusercontent.com/121920450/217755007-91bd97f2-b582-4d99-a70f-7658a1fbeffa.png">

**Reply by adrianeboyd** [reply]
Install spacy with only the command `pip install spacy==2.3.5` in a new virtual environment. It looks like your env has conflicting packages from a previous spacy v3 install.

**Reply by swagaths1** [reply]
yaaa.. this issue got resolved and  
pytest --pyargs spacy.tests.test_architectures is passing



_../../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/tensor_util.py:127
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/tensor_util.py:127: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    np.bool: SlowAppendBoolArrayToTensorProto,

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================================================== 1 passed, 16 warnings in 1.79s ==================_

**But other test cases it givings following error**
<img width="1728" alt="Screenshot 2023-02-09 at 2 11 34 PM" src="https://user-images.githubusercontent.com/121920450/217760824-b8893c05-89e2-463b-bfe0-4a221be2f945.png">

**Reply by rmitsch** [reply]
Please past error messages and code as text (formatted as code with backticks), not as image.
What's the `pytest` command you executed - `pytest --pyargs spacy.tests.test_ty`?

**Reply by swagaths1** [reply]
yes It was the same command

 pytest --pyargs spacy.tests.test_ty.py
================================================================================================================ test session starts ================================================================================================================
platform darwin -- Python 3.9.13, pytest-7.2.0, pluggy-1.0.0
rootdir: /Users/swagath/Documents/GitHub
plugins: hypothesis-6.68.0
collected 0 items                                                                                                                                                                                                                                   

================================================================================================================= warnings summary ==================================================================================================================
../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/pywrap_tensorflow.py:15
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/pywrap_tensorflow.py:15: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/google/protobuf/internal/containers.py:182
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/google/protobuf/internal/containers.py:182: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
    MutableMapping = collections.MutableMapping

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/google/protobuf/internal/containers.py:340
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/google/protobuf/internal/containers.py:340: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
    collections.MutableSequence.register(BaseContainer)

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:463
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:463: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
    _np_qint8 = np.dtype([("qint8", np.int8, 1)])

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:464
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:464: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
    _np_quint8 = np.dtype([("quint8", np.uint8, 1)])

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:465
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:465: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
    _np_qint16 = np.dtype([("qint16", np.int16, 1)])

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:466
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:466: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
    _np_quint16 = np.dtype([("quint16", np.uint16, 1)])

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:467
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:467: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
    _np_qint32 = np.dtype([("qint32", np.int32, 1)])

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:482
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:482: DeprecationWarning: `np.object` is a deprecated alias for the builtin `object`. To silence this warning, use `object` by itself. Doing this will not modify any behavior and is safe. 
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    (np.object, string),

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:483
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:483: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    (np.bool, bool),

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:502
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:502: DeprecationWarning: `np.object` is a deprecated alias for the builtin `object`. To silence this warning, use `object` by itself. Doing this will not modify any behavior and is safe. 
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    types_pb2.DT_STRING: np.object,

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:506
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:506: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    types_pb2.DT_BOOL: np.bool,

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:523
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:523: DeprecationWarning: `np.object` is a deprecated alias for the builtin `object`. To silence this warning, use `object` by itself. Doing this will not modify any behavior and is safe. 
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    types_pb2.DT_STRING_REF: np.object,

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:527
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/dtypes.py:527: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    types_pb2.DT_BOOL_REF: np.bool,

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/tensor_util.py:126
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/tensor_util.py:126: DeprecationWarning: `np.object` is a deprecated alias for the builtin `object`. To silence this warning, use `object` by itself. Doing this will not modify any behavior and is safe. 
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    np.object: SlowAppendObjectArrayToTensorProto,

../../opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/tensor_util.py:127
  /Users/swagath/opt/anaconda3/envs/test_transformer/lib/python3.9/site-packages/tensorflow/python/framework/tensor_util.py:127: DeprecationWarning: `np.bool` is a deprecated alias for the builtin `bool`. To silence this warning, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    np.bool: SlowAppendBoolArrayToTensorProto,

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================================================================================== 16 warnings in 2.45s ================================================================================================================
ERROR: module or package not found: spacy.tests.test_ty.py (missing __init__.py?)

**Reply by swagaths1** [reply]
Hi Team any updates?

**Reply by swagaths1** [reply]
Hi Team. any updates As it looking like I have installed spacy and its dependencies correctly. could you please guide me to fix this issue and run pytest smoothly .

**Reply by rmitsch** [reply]
- Did you install spaCy v2.3.5 in a new environment from scratch? 
- Note that it's `pytest --pyargs spacy.tests.test_ty`, not `pytest --pyargs spacy.tests.test_ty.py`. I.e. specify the _module_ name, not the file name.

In general I recommend following the [spaCy readme section on tests](https://github.com/explosion/spaCy#-run-tests): execute the tests with `python -m pytest --pyargs spacy`.

**Reply by swagaths1** [reply]
yes Now its working as expected. Thank you very much for your support throughout
